ledger-core/v2/runner/executionevent/event.go

```diff
@@ -132,89 +143,113 @@ type saveAsChild struct {
+func (e callConstructor) ParentObjectReference() reference.Global {
 	return e.parentObjectReference
 }
 
-func (e saveAsChild) ParentRequestReference() reference.Global {
+func (e callConstructor) ParentRequestReference() reference.Global {
 	return e.parentRequestReference
 }
 
-func (e saveAsChild) ConstructOutgoing(execution execution.Context) payload.VCallRequest {
-	panic(throw.NotImplemented())
+func (e callConstructor) ConstructVCallRequest(execution execution.Context) *payload.VCallRequest {
+	execution.Sequence++
+
+	return &payload.VCallRequest{
+		CallType:            payload.CTConstructor,
+		Caller:              e.parentObjectReference,
+		Callee:              reference.Global{},
+		CallSiteDeclaration: e.prototype,
+		CallSiteMethod:      e.constructor,
+		CallSequence:        execution.Sequence,
+		CallReason:          e.parentRequestReference,
+		KnownCalleeIncoming: reference.Global{},
+		CallOutgoing:        reference.Local{}, // must be filled in the caller
+		Arguments:           e.arguments,
+	}
 }
 
-type routeCall struct {
+func (e callConstructor) callConstructor() {}
+
+type callMethod struct {
 	parentRequestReference reference.Global
 	parentObjectReference  reference.Global
 
 	method    string
 	arguments []byte
 	object    reference.Global
 	prototype reference.Global
-	immutable bool
-	saga      bool
+	unordered bool
 }
 
-func (e routeCall) Saga() bool {
-	return e.saga
+func (e callMethod) Unordered() bool {
+	return e.unordered
 }
 
-func (e routeCall) Immutable() bool {
-	return e.immutable
-}
-
-func (e routeCall) Prototype() reference.Global {
+func (e callMethod) Prototype() reference.Global {
 	return e.prototype
 }
 
-func (e routeCall) Object() reference.Global {
+func (e callMethod) Object() reference.Global {
 	return e.object
 }
 
-func (e routeCall) Arguments() []byte {
+func (e callMethod) Arguments() []byte {
 	return e.arguments
 }
 
-func (e routeCall) Method() string {
+func (e callMethod) Method() string {
 	return e.method
 }
 
-func (e routeCall) ParentObjectReference() reference.Global {
+func (e callMethod) ParentObjectReference() reference.Global {
 	return e.parentObjectReference
 }
 
-func (e routeCall) ParentRequestReference() reference.Global {
+func (e callMethod) ParentRequestReference() reference.Global {
 	return e.parentRequestReference
 }
 
-type RouteCallBuilder interface {
-	SetSaga(isSaga bool) RouteCallBuilder
-	SetImmutable(isImmutable bool) RouteCallBuilder
+type callMethodBuilder interface {
+	SetImmutable(isImmutable bool) callMethodBuilder
 }
 
-func (e routeCall) ConstructOutgoing(execution execution.Context) payload.VCallRequest {
-	panic(throw.NotImplemented())
-}
+func (e callMethod) ConstructVCallRequest(execution execution.Context) *payload.VCallRequest {
+	execution.Sequence++
 
-func (e *routeCall) SetSaga(isSaga bool) RouteCallBuilder {
-	e.saga = isSaga
-	return e
+	var callFlags callflag.CallFlag
+	if e.unordered {
+		callFlags |= callflag.Unordered
```

Comment:
I would add auxiliary functions like it's done for StateRequestContentFlags

Reply:
No "set" functions in `StateRequestContentFlags `

Reply:
I've already added them. Nevertheless, CallFlags will be proper handled later after my PR #139 
In this PR they should just work